packages/next/src/cli/next-test.ts

```diff
@@ -0,0 +1,195 @@
+  } catch (err) {
+    // if that failed, then `directory` is not a valid path, so it must have meant to be the first item for `testRunnerArgs`
+    // @ts-expect-error directory is a string here since `getProjectDir` will succeed if its undefined
+    testRunnerArgs.unshift(directory)
+    // intentionally set baseDir to the resolved '.' path
+    baseDir = getProjectDir()
+  }
+
+  try {
+    // but, `baseDir` might not be a Next.js project directory, it could be a path-like argument for the test runner (i.e. `playwright test test/foo.spec.js`)
+    // if this succeeds, it means that `baseDir` is a Next.js project directory
+    nextConfig = await loadConfig(PHASE_PRODUCTION_BUILD, baseDir)
+  } catch (err) {
+    // if it doesn't, then most likely `baseDir` is not a Next.js project directory
+    // @ts-expect-error directory is a string here since `getProjectDir` will succeed if its undefined
+    testRunnerArgs.unshift(directory)
+    // intentionally set baseDir to the resolved '.' path
+    baseDir = getProjectDir()
+    nextConfig = await loadConfig(PHASE_PRODUCTION_BUILD, baseDir) // let this error bubble up if the `basePath` is still not a valid Next.js project
+  }
+
+  // set the test runner. priority is CLI option > next config > default 'playwright'
+  const configuredTestRunner =
+    options?.testRunner ?? // --test-runner='foo'
+    nextConfig.experimental.defaultTestRunner ?? // { experimental: { defaultTestRunner: 'foo' }}
+    'playwright'
+
+  if (!nextConfig.experimental.testProxy) {
+    return printAndExit(
+      `\`next experimental-test\` requires the \`experimental.testProxy: true\` configuration option.`
+    )
+  }
+
+  // execute test runner specific function
+  switch (configuredTestRunner) {
+    case 'playwright':
+      return runPlaywright(baseDir, nextConfig, testRunnerArgs)
+    default:
+      return printAndExit(
+        `Test runner ${configuredTestRunner} is not supported.`
+      )
+  }
+}
+
+async function checkRequiredDeps(
+  baseDir: string,
+  testRunner: SupportedTestRunners
+) {
+  const deps = await hasNecessaryDependencies(
+    baseDir,
+    requiredPackagesByTestRunner[testRunner]
+  )
+  if (deps.missing.length > 0) {
+    await installDependencies(baseDir, deps.missing, true)
+
+    const playwright = spawn(
+      path.join(baseDir, 'node_modules', '.bin', 'playwright'),
+      ['install'],
+      {
+        cwd: baseDir,
+        shell: false,
+        stdio: 'inherit',
+        env: {
+          ...process.env,
+        },
+      }
+    )
+
+    return new Promise((resolve, reject) => {
+      playwright.on('close', (c) => resolve(c))
+      playwright.on('error', (err) => reject(err))
+    })
+  }
+}
+
+async function runPlaywright(
+  baseDir: string,
+  nextConfig: NextConfigComplete,
+  testRunnerArgs: string[]
+) {
+  await checkRequiredDeps(baseDir, 'playwright')
+
+  const playwrightConfigFile = await findUp(
+    ['playwright.config.js', 'playwright.config.ts'],
```

Comment:
Does playwright also support `.mjs` or `.json` files for config?

Reply:
I don't think so. I can only find references to these two in the docs